app/main.py

```diff
@@ -1,3 +1,13 @@
 def cache(func):
-    # Write your code here
-    pass
+    cache_of_args = {}
+
+    def wrapper(*args, **kwargs):
+        if args not in cache_of_args:
+            res = func(*args, **kwargs)
+            cache_of_args[args] = res
+            print("Calculating new result")
+            return res
+        if args in cache_of_args.keys():
+            print("Getting from cache")
+            return cache_of_args[args]
```

Comment:
Duplication of the `return` statement, You can move it from `if`

Reply:
there is 2 different **return** from different **if**... i cant understand how correct properly

Reply:
you should return cached valure from `cache_of_args`

Reply:
sorry, i dont understand... i tried to run in debager and it works, values saves to Dict and then take and return it from cache(when the arguments repeats) or return and save result (when arguments was new)